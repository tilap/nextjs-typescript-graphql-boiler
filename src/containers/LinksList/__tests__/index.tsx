import { InMemoryCache, gql } from '@apollo/client';
import { MockedProvider } from '@apollo/client/testing';
import renderer from 'react-test-renderer'; // eslint-disable-line import/no-extraneous-dependencies
import Component from '..';

const createCache = (items: number) => {
  const cache = new InMemoryCache();
  cache.writeQuery({
    query: gql`
      query LinksList {
        links {
          id
          title
          url
        }
      }
    `,
    data: {
      links: new Array(items).fill('').map((_, i) => {
        return {
          __typename: 'Link',
          id: `link-${i}`,
          title: `link title #${i}`,
          url: `http://site-${i}.com`,
        };
      }),
    },
  });
  return cache;
};

describe('LinksList', () => {
  it('renders correctly when no item', () => {
    const component = renderer.create(
      <MockedProvider cache={createCache(0)}>
        <Component />
      </MockedProvider>,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('renders correctly  when some items', () => {
    const component = renderer.create(
      <MockedProvider cache={createCache(3)}>
        <Component />
      </MockedProvider>,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
