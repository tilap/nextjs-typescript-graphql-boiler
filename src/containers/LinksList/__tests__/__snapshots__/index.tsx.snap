// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinksList renders correctly  when some items 1`] = `
<div>
  <div>
    page-links:link-count
     
  </div>
  <ul>
    <li>
      <a
        href="http://site-0.com"
        rel="noreferrer"
        target="_blank"
      >
        link title #0
      </a>
    </li>
    <li>
      <a
        href="http://site-1.com"
        rel="noreferrer"
        target="_blank"
      >
        link title #1
      </a>
    </li>
    <li>
      <a
        href="http://site-2.com"
        rel="noreferrer"
        target="_blank"
      >
        link title #2
      </a>
    </li>
  </ul>
</div>
`;

exports[`LinksList renders correctly when no item 1`] = `
<div>
  <div>
    page-links:link-count
     
  </div>
</div>
`;
